File: workspaces/benchmarks/compute/mandelbrot/mandelbrot.py
# Mandelbrot Benchmark
# Calculate escape iterations for Mandelbrot set

def mandelbrot_escape(cr: float, ci: float, max_iter: int) -> int:
    zr = 0.0
    zi = 0.0
    
    for i in range(max_iter):
        zr2 = zr * zr
        zi2 = zi * zi
        
        if zr2 + zi2 > 4.0:
            return i
        
        zi = 2.0 * zr * zi + ci
        zr = zr2 - zi2 + cr
    
    return max_iter

def main():
    size = 2000
    max_iter = 50
    total_iter = 0
    
    for y in range(size):
        for x in range(size):
            cr = (2.0 * x / size) - 1.5
            ci = (2.0 * y / size) - 1.0
            total_iter += mandelbrot_escape(cr, ci, max_iter)
    
    print(f"Total iterations: {total_iter}")

if __name__ == "__main__":
    main()

File: workspaces/benchmarks/compute/mandelbrot/mandelbrot.rs
// Mandelbrot Benchmark
// Calculate escape iterations for Mandelbrot set

fn mandelbrot_escape(cr: f64, ci: f64, max_iter: i32) -> i32 {
    let mut zr = 0.0;
    let mut zi = 0.0;
    
    for i in 0..max_iter {
        let zr2 = zr * zr;
        let zi2 = zi * zi;
        
        if zr2 + zi2 > 4.0 {
            return i;
        }
        
        zi = 2.0 * zr * zi + ci;
        zr = zr2 - zi2 + cr;
    }
    
    max_iter
}

fn main() {
    let size = 2000;
    let max_iter = 50;
    let mut total_iter: i64 = 0;
    
    for y in 0..size {
        for x in 0..size {
            let cr = (2.0 * x as f64 / size as f64) - 1.5;
            let ci = (2.0 * y as f64 / size as f64) - 1.0;
            total_iter += mandelbrot_escape(cr, ci, max_iter) as i64;
        }
    }
    
    println!("Total iterations: {}", total_iter);
}
